plot-lib/plot/private/no-gui/plot2d-utils.rkt

```diff
@@ -68,23 +68,29 @@
 (define (plot-area area renderer-list)
   (send area start-plot)
   
-  (define legend-entries
-    (flatten-legend-entries
-     (for/list : (Listof (Treeof legend-entry)) ([rend  (in-list renderer-list)])
-       (match-define (renderer2d rend-bounds-rect _bf _tf label-proc render-proc) rend)
-       (send area start-renderer (if rend-bounds-rect
-                                     (rect-inexact->exact rend-bounds-rect)
-                                     (unknown-rect 2)))
-       (when render-proc (render-proc area))
-       (cond
-         [(and render-proc label-proc)
-          (define rect (send area get-bounds-rect))
-          (label-proc rect)]
-         [else empty]))))
+  (for ([rend  (in-list renderer-list)])
+    (match-define (renderer2d rend-bounds-rect _bf _tf label-proc render-proc) rend)
+    ;; next step could be moved into (when render-proc, but this generates different step files)
+    (send area start-renderer (if rend-bounds-rect
+                                  (rect-inexact->exact rend-bounds-rect)
+                                  (unknown-rect 2)))
+    (when render-proc
+      (render-proc area)))
   
   (send area end-renderers)
-  
-  (when (not (empty? legend-entries))
-    (send area draw-legend legend-entries))
-  
   (send area end-plot))
+
+(: get-legend-list (-> (Listof renderer2d) Rect (Listof legend-entry)))
+(define (get-legend-list renderer-list outer-rect)
```

Comment:
I think `get-legend-entries` is a better name for this (and other similar functions in this pull request).  `get-legend-list` seems to indicate that the plot has more than one legend.

Reply:
I will change it to `get-legend-entry-list`, to make the distinction with the `(Treeof legend-entry)`